### PR TITLE
CatalogClientImpl logging improvement

### PR DESCRIPTION
**Pull request by sergii-vostrikov**
PartitionsRequest, CatalogRequest and CatalogVersionRequest now can
generate readable keys by themselves.
CatalogClientImpl class now logs all its major steps.

Relates-to: OLPEDGE-547

Signed-off-by: Sergii Vostrikov <ext-sergii.vostrikov@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h [hunk 1]
```diff
@@ -88,7 +91,20 @@ class CatalogRequest final {
...
+  std::string CreateKey() const {
+    std::stringstream ss;
+
+    if (GetBillingTag()) {
+      ss << "$" << GetBillingTag().get();
+    }
+
+    ss << "^" << GetFetchOption();
+
+    return ss.str();
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h]: Should not be part of a public API

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h [hunk 2]
```diff
@@ -113,6 +109,26 @@ class DATASERVICE_READ_API CatalogVersionRequest final {
     fetch_option_ = fetchoption;
     return *this;
   }
+
+  /**
+   * @brief Creates readable format for the request.
+   * @param fetchoption enums
+   * @return a reference to the updated CatalogVersionRequest
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: Should not be exposed to the user.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h (hunk 1 above)
Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h]: naming adjusted

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h (hunk 2 above)
Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: description updated

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h [hunk 3]
```diff
@@ -88,7 +91,25 @@ class CatalogRequest final {
...
     return *this;
   }
- private:
+
+  /**
+   * @brief Creates readable format for the request.
+   * @param none
+   * @return string representation of the request
+   */
+  std::string CreateKey() const {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h]: This calls for a base class with abstract CreateKey()

Comment (andescu) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h]: Should be marked inline

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h [hunk 4]
```diff
@@ -113,6 +109,26 @@ class DATASERVICE_READ_API CatalogVersionRequest final {
     fetch_option_ = fetchoption;
     return *this;
   }
+
+  /**
+   * @brief Creates readable format for the request.
+   * @param none
+   * @return string representation of the request
+   */
+  std::string CreateKey() const {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: mark inline

Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: Compilers may decide to (un)inline the method regardless of the keyword.
I don't see the point of making it inline forcibly either.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h (hunk 3 above)
Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h]: There is no architectural usage base class now, thus abstract interface brings a negligible value.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h (hunk 4 above)
Comment (andescu) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-inline
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-inline
Not always the case, but for in-class defines this is true. Either mark all `inline` or none. Do not leave half states.

Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: Makes sense, I updated the definition.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h [hunk 5]
```diff
@@ -32,22 +33,19 @@ namespace dataservice {
...
    * @brief Mandatory for versioned layers; the beginning of the range of
    * versions you want to get (exclusive). By convention -1 indicates the
    * initial version before the first publication. After the first publication,
    * the catalog version is 0.
    * @return the start version.
    */
-  inline int64_t GetStartVersion() const {
-    return start_version_;
-  }
+  inline int64_t GetStartVersion() const { return start_version_; }
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: Are you sure this formatting is correct? For me, using the clang-format, it never formats to one-liners.

Comment (sergii-vostrikov) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogVersionRequest.h]: yes, that's clang-format's outcome

**Pull request merged**